File: src/main/java/mmo/core/netty/ChannelAdapter.java
package mmo.core.netty;

import io.netty.channel.ChannelHandlerContext;
import mmo.core.network.Channel;

import java.net.InetSocketAddress;

public final class ChannelAdapter implements Channel {
    private final ChannelHandlerContext channel;

    public ChannelAdapter(ChannelHandlerContext channel) {
        this.channel = channel;
    }

    @Override
    public Object id() {
        return channel.channel().id();
    }

    @Override
    public void write(Object message) {
        channel.writeAndFlush(message);
    }

    @Override
    public void close() {
        channel.close();
    }

    @Override
    public boolean isAlive() {
        return channel.channel().isActive();
    }

    @Override
    public InetSocketAddress address() {
        return (InetSocketAddress) channel.channel().remoteAddress();
    }
}

File: src/main/java/mmo/core/netty/ChannelAdapter.kt
package mmo.core.netty

import io.netty.channel.ChannelHandlerContext
import io.netty.channel.ChannelId
import mmo.core.network.Channel
import java.net.InetSocketAddress

class ChannelAdapter(private val channel: ChannelHandlerContext) : Channel {
    override fun id(): ChannelId {
        return channel.channel().id()
    }

    override fun write(message: Any?) {
        channel.writeAndFlush(message)
    }

    override fun close() {
        channel.close()
    }

    override val isAlive: Boolean
        get() = channel.channel().isActive

    override fun address(): InetSocketAddress? {
        return channel.channel().remoteAddress() as InetSocketAddress
    }
}